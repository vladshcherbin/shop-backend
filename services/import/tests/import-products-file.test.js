import { jest } from '@jest/globals'

jest.unstable_mockModule('@aws-sdk/s3-request-presigner', () => ({
  getSignedUrl: () => 'mocked-url'
}))

const { handler } = await import('../src/handlers/import-products-file')

test('Products file signed url is generated', async () => {
  const response = await handler({
    queryStringParameters: { name: 'test.csv' }
  })

  expect(response.statusCode).toBe(200)
  expect(response.body).toMatchSnapshot()
})

test('Error if name query parameter is missing', async () => {
  const response = await handler()

  expect(response.statusCode).toBe(500)
  expect(response.body).toMatchSnapshot()
})
