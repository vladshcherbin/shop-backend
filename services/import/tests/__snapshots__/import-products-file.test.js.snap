// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error if name query parameter is missing 1`] = `"{\\"message\\":\\"Internal server error\\"}"`;

exports[`Products file signed url is generated 1`] = `"{\\"url\\":\\"mocked-url\\"}"`;
